cincinnati/src/lib.rs

```diff
@@ -77,7 +79,11 @@ impl<'a> Iterator for NextReleases<'a> {
     fn next(&mut self) -> Option<Self::Item> {
         self.children
             .walk_next(self.dag)
-            .map(|(_, i)| self.dag.node_weight(i).unwrap())
+            .map(|(_, i)| self.dag.node_weight(i))
+            .unwrap_or_else(|| {
+                warn!("iterator shortened due to missing node weight");
```

Comment:
This one I'm actually wondering if you'd better like to either:
 * adjust to `Item = Result<...>` (not sure who the consumer it), or
 * turn this into an explicit panic (i.e. this is guaranteed not to happen), or
 * shorten the iterator and record a log entry (current path)

Reply:
The consumer was intended to be a client. I don't have a concrete Rust client though. That weight must always be present, so the panic is most correct but of course that is really bad form for a library. Let me take a look again today. Maybe we can just drop this function.